sdk/textanalytics/ai-text-analytics/review/ai-text-analytics.api.md

```diff
@@ -43,6 +57,83 @@ export type AnalyzeActionParameters<ActionName extends AnalyzeActionName> = {
     LanguageDetection: LanguageDetectionAction;
 }[ActionName];
 
+// @public
+export type AnalyzeBatchAction = {
+    actionName?: string;
+} & ((EntityLinkingAction & {
+    kind: "EntityLinking";
+}) | (EntityRecognitionAction & {
+    kind: "EntityRecognition";
+}) | (KeyPhraseExtractionAction & {
+    kind: "KeyPhraseExtraction";
+}) | (PiiEntityRecognitionAction & {
+    kind: "PiiEntityRecognition";
+}) | (SentimentAnalysisAction & {
+    kind: "SentimentAnalysis";
+}) | (HealthcareAction & {
+    kind: "Healthcare";
+}) | (CustomEntityRecognitionAction & {
+    kind: "CustomEntityRecognition";
+}) | (CustomSingleLabelClassificationAction & {
+    kind: "CustomSingleLabelClassification";
+}) | (CustomMultiLabelClassificationAction & {
+    kind: "CustomMultiLabelClassification";
+}));
+
+// @public
+export const AnalyzeBatchActionNames: {
+    readonly SentimentAnalysis: "SentimentAnalysis";
+    readonly EntityRecognition: "EntityRecognition";
+    readonly PiiEntityRecognition: "PiiEntityRecognition";
+    readonly KeyPhraseExtraction: "KeyPhraseExtraction";
+    readonly EntityLinking: "EntityLinking";
+    readonly Healthcare: "Healthcare";
+    readonly CustomEntityRecognition: "CustomEntityRecognition";
+    readonly CustomSingleLabelClassification: "CustomSingleLabelClassification";
+    readonly CustomMultiLabelClassification: "CustomMultiLabelClassification";
+};
+
+// @public
+export interface AnalyzeBatchOperationMetadata {
+    actionFailedCount: number;
+    actionInProgressCount: number;
+    actionSucceededCount: number;
+    createdOn: Date;
+    displayName?: string;
+    expiresOn?: Date;
+    lastModifiedOn: Date;
+    operationId: string;
+    status: State;
+}
+
+// @public
+export interface AnalyzeBatchOperationState extends PollOperationState<PagedAnalyzeBatchResult>, AnalyzeBatchOperationMetadata {
+}
+
+// @public
+export type AnalyzeBatchPoller = PollerLike<AnalyzeBatchOperationState, PagedAnalyzeBatchResult>;
+
+// @public
+export type AnalyzeBatchResult = (BatchActionResult<EntityLinkingResult> & ActionMetadata & {
```

Comment:
I see what you mean but `BatchActionResult` is a union and an interface can't extend a union because its members are not statically known. I can still break down the type using intersections, e.g.
```ts
type EntityRecognitionBatchResult = ActionMetadata & BatchActionResult<EntityRecognitionResult, "EntityRecognition">
```